openmdao/core/implicitcomponent.py

```diff
@@ -67,11 +67,22 @@ def _apply_nonlinear(self):
         """
         with self._unscaled_context(outputs=[self._outputs], residuals=[self._residuals]):
             with self._call_user_function('apply_nonlinear', protect_outputs=True):
+                args = [self._inputs, self._outputs, self._residuals]
                 if self._discrete_inputs or self._discrete_outputs:
-                    self.apply_nonlinear(self._inputs, self._outputs, self._residuals,
-                                         self._discrete_inputs, self._discrete_outputs)
+                    args += [self._discrete_inputs, self._discrete_outputs]
```

Comment:
Not sure why we didn't create `_apply_nonlinear_wrapper` here like we did with all the other api functions.

Reply:
Yeah I agree it's not totally consistent, but I did it that way because if I move all of that stuff into a wrapper function, there's almost nothing left of the original function so it didn't seem worth it.